File: 101-symmetric-tree/symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root.right and not root.left:
            return True
        if not root.right and root.left or root.right and not root.left:
            return False
        leftarr = []
        rightarr = []

        self.righthelper(root.right,rightarr,"root")
        self.lefthelper(root.left,leftarr,"root")
        return leftarr == rightarr
    def righthelper(self,root,arr,direction):
        
        if root.right:
            self.righthelper(root.right,arr,"r")

        arr.append(str(root.val)+direction)
        
        if root.left:
            self.righthelper(root.left,arr,"l")
        

    def lefthelper(self,root,arr,direction):
        if root.left:
            self.lefthelper(root.left,arr,"r")

        arr.append(str(root.val)+direction)
        

        if root.right:
            self.lefthelper(root.right,arr,"l")

File: 101-symmetric-tree/symmetric-tree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func isSymmetric(_ root: TreeNode?) -> Bool {
        
        return self.recursive(left_child: root!.left, right_child: root!.right);
    }

    func recursive(left_child left: TreeNode?, right_child right: TreeNode?) -> Bool{
        if(left == nil && right == nil){
            return true;
        }else if(left == nil || right == nil){
            return false;
        }
        if let left = left, let right = right{
            if left.val == right.val{
                return self.recursive(left_child : left.left, right_child : right.right) && self.recursive(left_child : left.right, right_child : right.left)
            }else{
                return false;
            }
        }
        return false;
    }
}